crypto/fipsmodule/rand/rand.c

```diff
@@ -539,6 +553,14 @@ void RAND_bytes_with_additional_data(uint8_t *out, size_t out_len,
 
   OPENSSL_cleanse(additional_data, 32);
 
+  if (1 == CRYPTO_get_snapsafe_active()) {
```

Comment:
Hmm this could be replaced by:
```suggestion
  if (1 == CRYPTO_get_snapsafe_generation(&snapsafe_generation_id)) {
```
and just always do the test. If snapsafe is not supported in runtime, the test will always pass.

Current impl saves a few cycles, which I guess is your motivation?